internal/services/storage/migration/account.go

```diff
@@ -204,3 +211,810 @@ func accountSchemaForV0AndV1() map[string]*pluginsdk.Schema {
 		},
 	}
 }
+
+func accountSchemaForV2() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.StorageAccountName,
+		},
+
+		"resource_group_name": azure.SchemaResourceGroupName(),
+
+		"location": azure.SchemaLocation(),
+
+		"account_kind": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(storage.Storage),
+				string(storage.BlobStorage),
+				string(storage.BlockBlobStorage),
+				string(storage.FileStorage),
+				string(storage.StorageV2),
+			}, true),
+			Default: string(storage.StorageV2),
```

Comment:
fwiw you don't need these for a state migration - only the Type and Required/Optional/Computed iirc

```suggestion
```